lib/android.js

```diff
@@ -203,17 +205,34 @@ exports.detect = function detect(config, opts, finished) {
 							queue.push(function (cb) {
 								findNDK(subdir, config, cb);
 							});
-							// Android Studio installs the NDK to ~/Library/Android/Sdk/ndk-bundle
-							fs.statSync(subdir).isDirectory() && fs.readdirSync(subdir).forEach(function (name) {
-								if (/ndk-bundle/i.test(name)) {
+
+							// Check under NDK side-by-side directory which contains multiple NDK installations.
+							// Each subfolder is named after the version of NDK installed under it. Favor newest version.
+							const ndkSideBySidePath = path.join(subdir, 'ndk');
+							if (fs.existsSync(ndkSideBySidePath) && fs.statSync(ndkSideBySidePath).isDirectory()) {
+								let fileNames = fs.readdirSync(ndkSideBySidePath);
+								fileNames.sort(versionStringSortComparer);
+								fileNames = fileNames.reverse();
```

Comment:
Maybe amend`versionStringSortComparer()` so the `reverse` isn't necessary?

Reply:
Comparers are typically assumed to be in ascending order. But I agree, this can be optimized. So, I added a generic `createReverseComparerFrom()` function which can reverse the sort of the given comparer.

(Alternatively, I could traverse the array backwards.)